### PR TITLE
[Lab 6] Add User Experience Features

### PR DESCRIPTION
**Pull request by Halil-Ibrahim-Kasapoglu**
This PR will cover work done related to lab 6 assignments.

```
Lab 6 description:

- Create a PR dedicated to this lab session where you will link work done in this lab: commits, wiki pages, issues, meeting notes, etc. Start the PR with a commit in a new branch. Then, add commits to the branch, and document all the work in the PR description and comments. The commits should include co-authors, if there are any, see [here](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors). The description and comments should include the names of the main contributors related to the linked work. The PR should be finalized today at 8:00 PM.
- Document 2 user stories of a group of people each that would use your system in a specific location (e.g. work, Taiwan), time (e.g. morning), and duration (e.g. 15 minutes). The users need not have names, images, etc., but they should have specific demographics. It's not a single person but a group of people with similar characteristics regarding the user story.
- Substantial commit(s) regarding UX design catered to the above-mentioned user stories.
- Substantial commit(s) regarding UI similarity between the web and mobile versions of the system.

_Please split the work among the group members._

```

Related Work:
- We need to create and document 2 user experience scenarios for enhancement:
    - Dark mode toggle option
    - Loading indicator on time-taking async operations 
- We should provide source codes for related UX enhancement on web and mobile clients.

Related Issues: 
- [x] #383
- [x] #382 
- [x] #384
- [x] #386
- [x] #387

### CODE REVIEW

**Comment by cemgungor1** [comment]
User scenarios are created and added to [wiki](https://github.com/bounswe/bounswe2024group2/wiki/User-Scenario-Groups). Issue can also be checked: [#383](https://github.com/bounswe/bounswe2024group2/issues/383)

**Comment by m-erkam** [comment]
The dark mode for mobile is initialized which fixes #386

**Comment by danzio19** [comment]
I have added dark mode functionality which fixes #382

**Comment by mahmutbugramert** [comment]
I added spinner to login page which fixes [#384](https://github.com/bounswe/bounswe2024group2/issues/384)

File: frontend/src/components/login/Login.js
Comment (Halil-Ibrahim-Kasapoglu) [inline frontend/src/components/login/Login.js:1]: It seems like this file is not properly merged and removes some of the additions by @danzio19 . I suspect the culprit is this [commit](https://github.com/bounswe/bounswe2024group2/commit/48124d67c4d3a11559529023580abab915dda26a). Can you take a look @mahmutbugramert

**Review by Halil-Ibrahim-Kasapoglu — COMMENTED** [review]
I have reviewed all the additions and left a comment regarding the changes on the frontend login page. It will be ready to go once that issue is resolved.

File: frontend/src/components/login/Login.js
Comment (mahmutbugramert) [reply frontend/src/components/login/Login.js:1]: Thank you for your feedback. I believe with [this ](https://github.com/bounswe/bounswe2024group2/pull/385/commits/1dac6b8778d296a5fd4f2e8ea0c9230318c2d967)commit. I restored the missing code.

**Review by Halil-Ibrahim-Kasapoglu — COMMENTED** [review]
Reviewed and confirmed that all changes are working as expected. Everything looks great. Approving and marking this as resolved. 👍

**Pull request merged**